apps/examples/testcase/le_tc/kernel/tc_sched.c

```diff
@@ -414,43 +416,42 @@ static void tc_sched_sched_lock_unlock(void)
 {
 	int ret_chk = ERROR;
 	int cntlock;
-	struct tcb_s *st_tcb = NULL;
+	int fd;
+	fd = tc_get_drvfd();
 
-	st_tcb = sched_self();
-	TC_ASSERT_NEQ("sched_self", st_tcb, NULL);
-
-	cntlock = st_tcb->lockcount;
+	cntlock = ioctl(fd, TESTIOC_GET_TCB_LOCKCOUNT, getpid());
```

Comment:
You'd better save pid in the local variable.
calling getpid many times is wasteful.